### PR TITLE
Update HolidayBase tests

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change

Update HolidayBase tests
  - Group tests by category
  - Use stubs instead of real entity objects
  - Use type specific asserts
  - Add `HolidayBase::__bool__()` tests

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Review by KJhellico — COMMENTED** [review]
This is very helpful! I had just redesigned the Canada used in these tests and was about to change them myself. 👍

File: tests/test_holiday_base.py [hunk 1]
```diff
@@ -9,851 +9,933 @@
...
+    market = "MS1"
+
+
+class MarketStub2(EntityStub):
+    market = "MS2"
+
+
+class TestArgs(unittest.TestCase):
+    def test_categories(self):
+        self.assertRaises(NotImplementedError, lambda: CountryStub1(categories={"HOME"}))
```
[the remark this answers is not in the document]
Comment (arkid15r) [reply tests/test_holiday_base.py]: Done

File: tests/test_holiday_base.py [hunk 2]
```diff
@@ -9,851 +9,933 @@
...
+
+    def test_ne(self):
+        hb_1 = CountryStub1()
+        hb_2 = CountryStub2()
+        hb_3 = CountryStub1(subdiv="Subdiv1")
+
+        self.assertNotEqual(hb_1, hb_2)
+        self.assertNotEqual(hb_1, {})
+        self.assertNotEqual(hb_2, {})
+
```
[the remark this answers is not in the document]
Comment (arkid15r) [reply tests/test_holiday_base.py]: Reworked this too

File: tests/test_holiday_base.py [hunk 3]
```diff
@@ -9,851 +9,933 @@
...
+            hb_combined.get_list("2021-08-10"),
+            ["Subdiv1 Custom Holiday", "Subdiv2 Custom Holiday"],
         )
+
+
+class TestGetNamed(unittest.TestCase):
+    def test_contains(self):
+        hb = CountryStub1(years=2022)
+        for name in ("New", "Year"):
+            self.assertListEqual(hb.get_named(name), [date(2022, 1, 1)])
```
[the remark this answers is not in the document]
Comment (arkid15r) [reply tests/test_holiday_base.py]: 👍

**Review by arkid15r — COMMENTED** [review]
PTAL

**Review by KJhellico — APPROVED** [review]
LGTM.

**Pull request merged**